internal/resolution/variablesources/bundle_test.go

```diff
@@ -4,19 +4,208 @@ import (
+				Properties: []property.Property{
+					{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "second-level-dependency", "version": "1.0.1"}`)},
+				},
+			},
+			CatalogName: fakeCatalogName,
+			InChannels:  []*catalogmetadata.Channel{&fakeChannel},
+		},
+
+		// Second level dependency that does not match requirements of the first level dependency.
+		"second-level-dependency.v2.0.0": {
+			Bundle: declcfg.Bundle{
+				Name:    "second-level-dependency.v2.0.0",
+				Package: "second-level-dependency",
+				Properties: []property.Property{
+					{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "second-level-dependency", "version": "2.0.0"}`)},
+				},
+			},
+			CatalogName: fakeCatalogName,
+			InChannels:  []*catalogmetadata.Channel{&fakeChannel},
+		},
+
+		// Package that is in a our fake catalog, but is not involved
+		// in this dependency chain. We need this to make sure that
+		// the testable function filters out unrelated bundles.
+		"uninvolved-package.v1.0.0": {
+			Bundle: declcfg.Bundle{
+				Name:    "uninvolved-package.v1.0.0",
+				Package: "uninvolved-package",
+				Properties: []property.Property{
+					{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "uninvolved-package", "version": "1.0.0"}`)},
+				},
+			},
+			CatalogName: fakeCatalogName,
+			InChannels:  []*catalogmetadata.Channel{&fakeChannel},
+		},
+	}
+
+	allBundles := make([]*catalogmetadata.Bundle, 0, len(bundleSet))
+	for _, bundle := range bundleSet {
+		allBundles = append(allBundles, bundle)
+	}
+	requiredPackages := []*olmvariables.RequiredPackageVariable{
+		olmvariables.NewRequiredPackageVariable("test-package", []*catalogmetadata.Bundle{
+			bundleSet["first-level-dependency.v1.0.0"],
+		}),
+	}
+	installedPackages := []*olmvariables.InstalledPackageVariable{
+		olmvariables.NewInstalledPackageVariable("test-package", []*catalogmetadata.Bundle{
+			bundleSet["first-level-dependency.v1.0.0"],
+		}),
+	}
+
+	bundles, err := variablesources.MakeBundleVariables(allBundles, requiredPackages, installedPackages)
+	require.NoError(t, err)
+
+	// Each dependency must have a variable.
+	// Dependencies from the same package must be sorted by version
+	// with higher versions first.
+	expectedVariables := []*olmvariables.BundleVariable{
+		olmvariables.NewBundleVariable(
+			bundleSet["first-level-dependency.v1.0.0"],
+			[]*catalogmetadata.Bundle{
+				bundleSet["second-level-dependency.v1.0.1"],
+				bundleSet["second-level-dependency.v1.0.0"],
+			},
+		),
+		olmvariables.NewBundleVariable(
+			bundleSet["second-level-dependency.v1.0.1"],
+			nil,
+		),
+		olmvariables.NewBundleVariable(
+			bundleSet["second-level-dependency.v1.0.0"],
+			nil,
+		),
+	}
+	gocmpopts := []cmp.Option{
+		cmpopts.IgnoreUnexported(sync.RWMutex{}),
+		cmp.AllowUnexported(
+			sync.RWMutex{},
+			olmvariables.BundleVariable{},
+			input.SimpleVariable{},
+			constraint.DependencyConstraint{},
+			catalogmetadata.Bundle{},
```

Comment:
I think we can put this into `IgnoreUnexported`. Unexported fields are internal state and exported fields will be enough for comparions.

Reply:
I think this will also spare us from using `IgnoreFields`.